### PR TITLE
Darts: Flatten sub-cases to single level - to simplify code generation

### PR DESCRIPTION
**Pull request by macta**

### CODE REVIEW

File: exercises/darts/canonical-data.json [hunk 1]
```diff
@@ -1,101 +1,99 @@
 {
   "exercise": "darts",
-  "version": "1.2.0",
+  "version": "1.2.1",
```
Comment (SleeplessByte) [inline exercises/darts/canonical-data.json]: This is a breaking change. Should be 2.0.0

Comment (macta) [reply exercises/darts/canonical-data.json]: I don't think there is anything for me to do, as you've increased the version number right (and I'll remember for future that while functionaly equivalent - its structurally different)

Comment (SleeplessByte) [reply exercises/darts/canonical-data.json]: I've not increased it?

It really should be 2.0.0.

Comment (macta) [reply exercises/darts/canonical-data.json]: Ah yes - momentary mental freeze, will resubmit

**Comment by SleeplessByte** [comment]
Thank you @macta.

I've merged this as we had three (3) approvals.

**Pull request merged**